### ISSUE TITLE
Simplify acessing response body

### ISSUE THREAD

**Issue by clue** [reactphp/http-client#48]
Accessing the HTTP response body is likely one of the main use cases for this library, however it's actually unnecessarily cumbersome.

This has come up several times already, so we way want to look into providing a simpler API.

Possibly related to #41.

**Comment by WyriHaximus** [comment]
While try I like to keep streaming without storing data posssible, maybe #29 is a way to achieve this? Now the question is which one do we want to be the default. I personally think that easy access to the response body should be the default, and streaming it in should be a secondary option

**Comment by clue** [comment]
This likely depends on #70 in order to fix stream semantics first.

We may then provide normal stream buffering routines and should probably add an example here.

Once this is in, we may want to look into #41 to implement PSR-7 message body support.

**Comment by clue** [comment]
Minor progress update: #70 has been implemented, which means that one can now use ReactPHP's normal stream primitives to buffer the whole request body:


```php
$request = $client->request('GET', 'https://google.com/');
$request->on('response', function (Response $response) {
    \React\Promise\Stream\buffer($response)->then(function ($body) {
        echo $body;
    }, 'printf');
});
$request->end();
```

We're still working on a simpler API for #41 :+1:

**Comment by slince** [comment]
@clue  It seems the function is not exists. .

**Comment by jsor** [comment]
@slince This function is from the [react/promise-stream](https://github.com/reactphp/promise-stream) package.

```bash
composer req react/promise-stream
```

**Comment by slince** [comment]
@jsor  thks

**Comment by clue** [comment]
I'm closing this as it is already supported by the new HTTP client that has been merged into `react/http` via reactphp/http#368 :shipit: You should upgrade as per #152 :+1:
